File: tutorials/mpi/examples/Makefile.MPI.f
##############################################################################
# FILE: Makefile.MPI.f
# DESCRIPTION:
#   Makefile for all IBM MPI Fortran example codes
# AUTHOR: Blaise Barney
# LAST REVISED:  05/25/17
##############################################################################

#Uncomment your choice of compiler
F90    =    mpif90
#F90    =    mpiifort
#F90    =    mpigfortran
#F90    =    mpipgf90
CC 	=   icc

#Add your choice of flags
FLAGS   =   -O2 -w

.PHONY:	cleanup

all:    mpi_array       \
        mpi_mm          \
        mpi_pi_send     \
        mpi_pi_reduce   \
        mpi_wave        \
        mpi_heat2D      \
        mpi_latency     \
        mpi_bandwidth   \
        mpi_prime       \
        mpi_ping        \
        mpi_ringtopo    \
        mpi_scatter     \
        mpi_contig      \
        mpi_vector      \
        mpi_indexed     \
        mpi_struct      \
        mpi_group       \
        mpi_cartesian	\
	cleanup   

cleanup:  
	/bin/rm -rf *.o     

clean:  
	/bin/rm -rf     \
	mpi_array       \
	mpi_mm          \
	mpi_pi_send     \
	mpi_pi_reduce   \
	mpi_wave        \
	mpi_heat2D      \
	mpi_latency     \
	mpi_bandwidth   \
	mpi_prime       \
	mpi_ping        \
	mpi_ringtopo    \
	mpi_scatter     \
	mpi_contig      \
	mpi_vector      \
	mpi_indexed     \
	mpi_struct      \
	mpi_group       \
	mpi_cartesian   \
	initial.dat     \
	final.dat       \
        *.o


mpi_array:  mpi_array.f
	$(F90) $(FLAGS) mpi_array.f -o mpi_array

mpi_mm:  mpi_mm.f
	$(F90) $(FLAGS) mpi_mm.f -o mpi_mm

mpi_pi_send:  mpi_pi_send.f
	$(F90) $(FLAGS) mpi_pi_send.f -o mpi_pi_send

mpi_pi_reduce:  mpi_pi_reduce.f 
	$(F90) $(FLAGS) mpi_pi_reduce.f -o mpi_pi_reduce

mpi_wave:  mpi_wave.f draw_wavef.o
	$(F90) $(FLAGS) mpi_wave.f draw_wavef.o -o mpi_wave -lX11
draw_wavef.o:  draw_wavef.c
	$(CC) -c draw_wavef.c 

mpi_heat2D:  mpi_heat2D.f draw_heatf.o
	$(F90) $(FLAGS) mpi_heat2D.f draw_heatf.o -o mpi_heat2D -lX11
draw_heatf.o:  draw_heatf.c
	$(CC) -c $(FLAGS) draw_heatf.c 

mpi_latency:  mpi_latency.f
	$(F90) $(FLAGS) mpi_latency.f -o mpi_latency

mpi_bandwidth:  mpi_bandwidth.f 
	$(F90) $(FLAGS) mpi_bandwidth.f -o mpi_bandwidth

mpi_prime:  mpi_prime.f 
	$(F90) $(FLAGS) mpi_prime.f -o mpi_prime

mpi_ping:  mpi_ping.f 
	$(F90) $(FLAGS) mpi_ping.f -o mpi_ping

mpi_ringtopo:  mpi_ringtopo.f 
	$(F90) $(FLAGS) mpi_ringtopo.f -o mpi_ringtopo

mpi_scatter:  mpi_scatter.f 
	$(F90) $(FLAGS) mpi_scatter.f -o mpi_scatter

mpi_contig:  mpi_contig.f 
	$(F90) $(FLAGS) mpi_contig.f -o mpi_contig

mpi_vector:  mpi_vector.f 
	$(F90) $(FLAGS) mpi_vector.f -o mpi_vector

mpi_indexed:  mpi_indexed.f 
	$(F90) $(FLAGS) mpi_indexed.f -o mpi_indexed

mpi_struct:  mpi_struct.f 
	$(F90) $(FLAGS) mpi_struct.f -o mpi_struct

mpi_group:  mpi_group.f 
	$(F90) $(FLAGS) mpi_group.f -o mpi_group

mpi_cartesian:  mpi_cartesian.f 
	$(F90) $(FLAGS) mpi_cartesian.f -o mpi_cartesian

File: tutorials/mpi/examples/Makefile.MPI.c
##############################################################################
# FILE: Makefile.MPI.c
# DESCRIPTION:
#   Makefile for all MPI C Language example codes
# AUTHOR: Blaise Barney
# LAST REVISED:  05/25/17
###############################################################################

#Uncomment your choice of compiler
CC    =   mpicc
#CC    =   mpiicc
#CC    =   mpigcc
#CC    =   mpipgcc

#Add your choice of flags
FLAGS   =   -O2 -w

all:    mpi_array       \
        mpi_bandwidth   \
        mpi_cartesian   \
        mpi_contig      \
        mpi_group       \
        mpi_heat2D      \
        mpi_indexed     \
        mpi_mm          \
        mpi_pi_reduce   \
        mpi_pi_send     \
        mpi_ping        \
        mpi_prime       \
        mpi_ringtopo    \
        mpi_scatter     \
        mpi_struct      \
        mpi_latency     \
        mpi_vector      \
        mpi_wave        


clean:  
	/bin/rm -rf     \
	initial.dat     \
	final.dat       \
	mpi_array       \
	mpi_bandwidth   \
	mpi_cartesian   \
	mpi_contig      \
	mpi_group       \
	mpi_heat2D      \
	mpi_indexed     \
	mpi_mm          \
	mpi_pi_reduce   \
	mpi_pi_send     \
	mpi_ping        \
	mpi_prime       \
	mpi_ringtopo    \
	mpi_scatter     \
	mpi_struct      \
	mpi_latency     \
	mpi_vector      \
	mpi_wave        \
        *.o
        


mpi_array:  mpi_array.c
	$(CC) $(FLAGS) mpi_array.c -o mpi_array

mpi_bandwidth:  mpi_bandwidth.c 
	$(CC) $(FLAGS) mpi_bandwidth.c -o mpi_bandwidth

mpi_cartesian:  mpi_cartesian.c 
	$(CC) $(FLAGS) mpi_cartesian.c -o mpi_cartesian

mpi_contig:  mpi_contig.c 
	$(CC) $(FLAGS) mpi_contig.c -o mpi_contig

mpi_group:  mpi_group.c 
	$(CC) $(FLAGS) mpi_group.c -o mpi_group

mpi_heat2D:  mpi_heat2D.c draw_heat.c
	$(CC) $(FLAGS) mpi_heat2D.c draw_heat.c -o mpi_heat2D -lX11

mpi_indexed:  mpi_indexed.c 
	$(CC) $(FLAGS) mpi_indexed.c -o mpi_indexed

mpi_mm:  mpi_mm.c
	$(CC) $(FLAGS) mpi_mm.c -o mpi_mm

mpi_pi_reduce:  mpi_pi_reduce.c
	$(CC) $(FLAGS) mpi_pi_reduce.c -o mpi_pi_reduce

mpi_pi_send:  mpi_pi_send.c
	$(CC) $(FLAGS) mpi_pi_send.c -o mpi_pi_send

mpi_ping:  mpi_ping.c 
	$(CC) $(FLAGS) mpi_ping.c -o mpi_ping

mpi_prime:  mpi_prime.c 
	$(CC) $(FLAGS) mpi_prime.c -o mpi_prime -lm

mpi_ringtopo:  mpi_ringtopo.c 
	$(CC) $(FLAGS) mpi_ringtopo.c -o mpi_ringtopo

mpi_scatter:  mpi_scatter.c 
	$(CC) $(FLAGS) mpi_scatter.c -o mpi_scatter

mpi_struct:  mpi_struct.c 
	$(CC) $(FLAGS) mpi_struct.c -o mpi_struct

mpi_latency:  mpi_latency.c 
	$(CC) $(FLAGS) mpi_latency.c -o mpi_latency

mpi_wave:  mpi_wave.c draw_wave.c
	$(CC) $(FLAGS) mpi_wave.c draw_wave.c -o mpi_wave -lX11 -lm

mpi_vector:  mpi_vector.c 
	$(CC) $(FLAGS) mpi_vector.c -o mpi_vector




